src/cascadia/UnitTests_Control/ControlCoreTests.cpp

```diff
@@ -499,4 +501,68 @@ namespace ControlUnitTests
             VERIFY_ARE_EQUAL(expectedEnd, end);
         }
     }
+
+    void ControlCoreTests::TestSimpleClickSelection()
+    {
+        // Create a simple selection with the mouse, then click somewhere else,
+        // and confirm the selection got updated.
+
+        auto [settings, conn] = _createSettingsAndConnection();
+        Log::Comment(L"Create ControlCore object");
+        auto core = createCore(*settings, *conn);
+        VERIFY_IS_NOT_NULL(core);
+        _standardInit(core);
+
+        // Here, we're using the UpdateSelectionMarkers as a stand-in to check
+        // if the selection got updated with the renderer. Standing up a whole
+        // dummy renderer for this test would be not very ergonomic. Instead, we
+        // are relying on ControlCore::_updateSelectionUI both
+        // TriggerSelection'ing and also rasing this event
+        bool expectedSelectionUpdate = false;
+        bool gotSelectionUpdate = false;
+        core->UpdateSelectionMarkers([&](auto&& /*sender*/, auto&& /*args*/) {
+            VERIFY_IS_TRUE(expectedSelectionUpdate);
+            expectedSelectionUpdate = false;
+            gotSelectionUpdate = true;
+        });
+
+        auto needToCopy = false;
+        expectedSelectionUpdate = true;
+        core->LeftClickOnTerminal(til::point{ 1, 1 },
+                                  1,
+                                  false,
+                                  true,
+                                  false,
+                                  needToCopy);
+
+        VERIFY_IS_TRUE(core->HasSelection());
+        {
+            const auto& start = core->_terminal->GetSelectionAnchor();
+            const auto& end = core->_terminal->GetSelectionEnd();
+            const til::point expectedStart{ 1, 1 };
+            const til::point expectedEnd{ 1, 1 };
+            VERIFY_ARE_EQUAL(expectedStart, start);
+            VERIFY_ARE_EQUAL(expectedEnd, end);
+        }
+        VERIFY_IS_TRUE(gotSelectionUpdate);
+
+        expectedSelectionUpdate = true;
+        core->LeftClickOnTerminal(til::point{ 1, 2 },
+                                  1,
+                                  false,
+                                  true,
+                                  false,
+                                  needToCopy);
+
+        VERIFY_IS_TRUE(core->HasSelection());
+        {
+            const auto& start = core->_terminal->GetSelectionAnchor();
+            const auto& end = core->_terminal->GetSelectionEnd();
+            const til::point expectedStart{ 1, 1 };
+            const til::point expectedEnd{ 1, 2 };
+            VERIFY_ARE_EQUAL(expectedStart, start);
+            VERIFY_ARE_EQUAL(expectedEnd, end);
+        }
+        VERIFY_IS_TRUE(gotSelectionUpdate);

```

Comment:
I don't actually understand how this selection test is related to click-to-reposition...

Reply:
It's actually not! But I broke that selection rendering thing in that last bug bash, in an earlier commit in this PR. This test I wrote to make sure I un-broke it, and won't re-break it in the future.

Reply:
Oh, I see. I thought it was a _render lock_ bug, so i didn't put the two together